ui/apps/platform/src/Containers/NetworkGraph/components/DeploymentSelector.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useCallback } from 'react';
+import { Select, SelectGroup, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+import { NamespaceWithDeployments } from 'hooks/useFetchNamespaceDeployments';
+import { DeploymentIcon } from '../common/NetworkGraphIcons';
+
+type DeploymentSelectorProps = {
+    deploymentsByNamespace: NamespaceWithDeployments[];
+    selectedDeployments: string[];
+    searchFilter: Partial<Record<string, string | string[]>>;
```

Comment:
I think there is a `SearchFilter` type already defined in `search.ts`. You can reuse that here